src/order/basic.lean

```diff
@@ -612,6 +614,37 @@ or_iff_not_imp_left.2 $ λ h,
   ⟨λ a ha₁, le_of_not_gt $ λ ha₂, h ⟨a, ha₁, ha₂⟩,
     λ a ha₂, le_of_not_gt $ λ ha₁, h ⟨a, ha₁, ha₂⟩⟩
 
+/-- Order equipped with a sensible successor function. -/
+class succ_order (α : Type u) [preorder α] :=
+(succ : α → α)
+(lt_iff_succ_le : ∀ a b, a < b ↔ succ a ≤ b)
+(lt_succ_iff_le : ∀ a b, a < succ b ↔ a ≤ b)
+
+open succ_order
+
+lemma lt_iff_succ_le {α : Type u} [preorder α] [succ_order α] {a b : α} :
+  a < b ↔ succ a ≤ b :=
+lt_iff_succ_le a b
+
+lemma lt_succ_iff_le {α : Type u} [preorder α] [succ_order α] {a b : α} :
+  a < succ b ↔ a ≤ b :=
+lt_succ_iff_le a b
```

Comment:
The following (untested) lemmas are used a lot, so they seem like an obvious addition:
```suggestion
lt_succ_iff_le a b

lemma lt_succ {α : Type u} [preorder α] [succ_order α] (a : α) :
  a < succ a :=
lt_succ_iff_le.mpr (le_refl a)

lemma bot_lt_succ {α : Type u} [bot_order α] [succ_order α] (a : α) :
  ⊥ < succ a :=
lt_succ_iff_le.mpr (bot_le a)
```